editor/src/clj/editor/breakpoints_view.clj

```diff
@@ -0,0 +1,526 @@
+  (let [breakpoints (:breakpoints state)
+        action-all-fn (fn [action-fn _]
+                        (g/with-auto-evaluation-context evaluation-context
+                          (set-regions-with-action! swap-state project evaluation-context breakpoints breakpoints action-fn)))]
+    {:fx/type fx.h-box/lifecycle
+     :id "breakpoints-tool-bar"
+     :anchor-pane/top 0
+     :anchor-pane/left 0
+     :anchor-pane/right 0
+     :pref-height toolbar-height
+     :alignment :center-left
+     :spacing 10
+     :children [{:fx/type fx.button/lifecycle
+                 :id "breakpoints-enable-all"
+                 :text (localization-state (localization/message "breakpoints.button.enable-all"))
+                 :on-action (fn/partial action-all-fn #(update-breakpoints-enabled-state %1 %2 true))}
+                {:fx/type fx.button/lifecycle
+                 :id "breakpoints-disable-all"
+                 :text (localization-state (localization/message "breakpoints.button.disable-all"))
+                 :on-action (fn/partial action-all-fn #(update-breakpoints-enabled-state %1 %2 false))}
+                {:fx/type fx.button/lifecycle
+                 :id "breakpoints-toggle-all"
+                 :text (localization-state (localization/message "breakpoints.button.toggle-all"))
+                 :on-action (fn/partial action-all-fn toggle-breakpoints-enabled)}
+                {:fx/type fx.button/lifecycle
+                 :id "breakpoints-remove-all"
+                 :text (localization-state (localization/message "breakpoints.button.remove-all"))
+                 :on-action (fn/partial action-all-fn #(vec (remove (set %2) %1)))}]}))
+
+(defn- icon-button [icon-path scale classes on-action-event]
+  {:fx/type fx.button/lifecycle
+   :style-class (into ["icon-button"] classes)
+   :graphic {:fx/type fx.stack-pane/lifecycle
+             :min-width 20
+             :min-height 20
+             :alignment :center
+             :children [{:fx/type fx.svg-path/lifecycle
+                         :content icon-path
+                         :scale-x scale
+                         :scale-y scale
+                         :style-class ["icon-button-graphic"]}]}
+   :on-action on-action-event})
+
+(defn- table-row-factory [open-resource-fn breakpoints idx]
+  (when-let [bp (get breakpoints idx)]
+    {:on-mouse-clicked
+     (fn [event]
+       (let [^MouseEvent e event
+             {:keys [resource row]} bp]
+         (when (and (= MouseButton/PRIMARY (.getButton e))
+                    (= 2 (.getClickCount e)))
+           (open-resource-fn resource (inc row)))))}))
+
+(defn- column-enabled-cell-factory [project swap-state breakpoints idx]
+  (when-let [breakpoint (get breakpoints idx)]
+    {:style-class ["enabled-cell"]
+     :alignment :center
+     :graphic
+     {:fx/type fx.check-box/lifecycle
+      :selected (:enabled breakpoint)
+      :on-selected-changed
+      (fn [_]
```

Comment:
Tip: You can supply a name symbol to `fns` to have them show up with nice names in callstacks: `(fn on-selected-changed [_] ...)`. Useful for callbacks like this!